src/runtime_src/xocl/api/xlnx/clPollStream.cpp

```diff
@@ -46,11 +46,8 @@ clPollStream(cl_stream                 stream,
 	cl_int*                         errcode_ret)
 {
   validOrError(stream,completions,min,max,actual,timeout,errcode_ret);
-
-  printf("clPollStream() -> xocl(stream)->poll_stream.\n");
-
-  xocl::xocl(stream)->poll_stream(completions,min,max,actual,timeout);
-  xocl::assign(errcode_ret,CL_SUCCESS);
+  int ret = xocl::xocl(stream)->poll_stream(completions,min,max,actual,timeout);
+  xocl::assign(errcode_ret, ret);
```

Comment:
poll_stream should throw on error, not return an error code.
the original code was correct.